maggy/optimizer/abstractoptimizer.py

```diff
@@ -24,9 +27,26 @@ def __init__(self):
         self.trial_store = None
         self.final_store = None
         self.direction = None
+        self.pruner = None
+
+    def initialize(self, exp_dir):
```

Comment:
can we do it the other way around, make this a private method e.g. `_initialize` and keep the developer api a public method.
```suggestion
    def _initialize(self, exp_dir):
```
and move to bottom

Reply:
sure

Reply:
we have to change the method then aswell in l.275 of experimentdriver to ._initialize()

Reply:
yes that's fine, I'd just like to keep the API that users see the way it is.